### PR TITLE
#[UIC-1912]: added single selection highlight support in line chart on click

### PR DESCRIPTION
**Pull request by jitendra-kumawat**
# CATEGORY
 Added support in line chart on click for single selection
-- highlight selection
-- remove highlight selection on new item click or self click
-- publish selected items 

Choose one

- [x] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screenshot 2019-08-01 at 11 20 09 AM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@arpit-agarwal

### CODE REVIEW

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js [hunk 1]
```diff
@@ -340,6 +340,26 @@ function nvd3Vis(element, props) {
         }
         chart.lines.dispatch.on('elementClick', function (e) {
           if (tableFilter) {
+            const removeSelection = (selection && selection.seriesIndex == e.seriesIndex && selection.pointIndex == e.pointIndex);
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/nvd3/NVD3Vis.js]: We can move this line more close to place where we need this. Currently it is creation . confusion with highlight

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js [hunk 2]
```diff
@@ -361,13 +381,13 @@ function nvd3Vis(element, props) {
...
+              xFieldVal = removeSelection ? null : getXAxisFieldVal(xField, e.point.x, columns)
+              xValueChanged = removeSelection || selection ? selection['point'].x != e.point.x : true;
             }
 
             if (yField != undefined && e.point) {
-              yFieldVal = e.point.y;
-              yValueChanged = selection ? selection['point'].y != yFieldVal  : true;
+              yFieldVal = removeSelection ? null : e.point.y;
+              yValueChanged = removeSelection || selection ? selection['point'].y != yFieldVal : true;
             }
```
Comment (arpit-agarwal) [inline superset/assets/src/visualizations/nvd3/NVD3Vis.js]: can we introduce somthing like below
```
if(removeselection) {
xValue = yValue = null;
xChanged = yChanged = true;
} else {
// existing code for determining xValue, yValue, changed flag
}
```

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js (hunk 1 above)
Comment (jitendra-kumawat) [reply superset/assets/src/visualizations/nvd3/NVD3Vis.js]: done

File: superset/assets/src/visualizations/nvd3/NVD3Vis.js (hunk 2 above)
Comment (jitendra-kumawat) [reply superset/assets/src/visualizations/nvd3/NVD3Vis.js]: done

**Pull request merged**